spec/unit/provider/user/pw_spec.rb

```diff
@@ -81,6 +81,21 @@
       provider.create
     end
 
+    it "should call execute with sensitive true when the password property is set" do
+      Puppet::Util::Log.level = :debug
+      resource[:password] = "abc123"
+      expect(provider).to receive(:execute).with(kind_of(Array), hash_including(sensitive: true))
+      expect(provider).to receive(:set_password_internal).with("abc123")
```

Comment:
Can we stub the `popen` call instead instead of the `get|set_password_internal` ? Otherwise we're stubbing out code like `current_passline.chomp.split(':')[1] if current_passline`? Alternatively move that code out of the internal method